recipes/gklib/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+from conan.tools.cmake import CMakeToolchain, CMake
+from conan.tools.files import apply_conandata_patches
+from conan.tools.layout import cmake_layout
+import os
+
+required_conan_version = ">=1.42.2"
+
+
+class GKlibConan(ConanFile):
+    name = "gklib"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/KarypisLab/GKlib"
+    description = "A library of various helper routines and frameworks" \
+                  " used by many of the lab's software"
+    topics = ("karypislab")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    @property
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def validate(self):
+        if self.options.shared:
+            raise ConanInvalidConfiguration(
+                f"{self.name} {self.version} couldn't be built in shared configuration")
+
+    def layout(self):
+        cmake_layout(self)
+        self.folders.source = "{}-{}".format(self.name, self.version)
+
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            strip_root=True,
+            destination=self.folders.source
+        )
+
+    def generate(self):
+        toolchain = CMakeToolchain(self)
+        toolchain.variables["BUILD_SHARED_LIBS"] = self.options.shared
+        toolchain.variables["ASSERT"] = self.settings.build_type == "Debug"
+        toolchain.variables["ASSERT2"] = self.settings.build_type == "Debug"
+        toolchain.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared
+        toolchain.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        CMake(self).install()
+        self.copy("LICENSE.txt", dst="licenses")
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "GKlib"
+        self.cpp_info.names["cmake_find_package_multi"] = "GKlib"
+        self.cpp_info.set_property("cmake_file_name", "GKlib")
+        self.cpp_info.set_property("cmake_target_name", "GKlib")
+        self.cpp_info.set_property("cmake_find_mode", "both")
```

Comment:
Taking a quick look at https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules, I dont see it listed as a module.

Is the module provided somewhere else? 🤔 I am not familiar with this project so I'd appreciate any extra context!

Reply:
I couldn't find any usages of "find_package(GKlib" in the wild. Karypislab repositories (https://github.com/KarypisLab) get GKlib as submodule

METIS for example uses source files from GKlib instead of linking to it
https://github.com/KarypisLab/METIS/blob/22008804e8c9b78893ae10a94c0d8b4b592438b4/libmetis/CMakeLists.txt#L6

So I'll need to add find_package there through patch

Reply:
### Optional commentary

Nit but if it's not required better to remove it

```suggestion
        self.cpp_info.set_property("cmake_find_mode", "both")
```

Reply:
Please remove all these lines.